src/net/primitives.c

```diff
@@ -195,8 +196,15 @@ z_result_t _z_write(_z_session_t *zn, const _z_keyexpr_t *keyexpr, const _z_byte
                     z_priority_t priority, bool is_express, const _z_timestamp_t *timestamp,
                     const _z_bytes_t *attachment, z_reliability_t reliability, const _z_source_info_t *source_info) {
     z_result_t ret = _Z_RES_OK;
-    _z_network_message_t msg;
     _z_qos_t qos = _z_n_qos_make(is_express, cong_ctrl == Z_CONGESTION_CONTROL_BLOCK, priority);
+
+    bool handled_locally = _z_session_deliver_push_locally(zn, keyexpr, payload, encoding, kind, qos, timestamp,
+                                                           attachment, reliability, source_info);
+    if (handled_locally && !_z_session_has_remote_targets(zn)) {
```

Comment:
I don't understand this condition. If there are no remote targets, then why does `handled_locally` matter when deciding whether to publish to the network or not?

Reply:
Well, if `handled_locally == false` we fall back to the old behaviour, so it's unchanged: every `_z_write` went through `_z_send_n_msg`, even if nobody was listening (or we don't know they are listening, for example targets built without interest support). If `handled_locally == true` we skip the network path, otherwise there would be no sense in these changes. Is it okay?